docs/extending-and-experimenting.md

```diff
@@ -147,3 +147,130 @@ subscribe to the result of `executeQuery`, keep track
 of the unsubscription (`teardown`) and update
 some state. This all can be reapplied when you write your
 own APIs.
+
+## Making your own Exchange
+
+As with any library, you're going to find yourself wanting to change (or adapt) the way things work.
+Since day one, with `urql`, we have aimed to make adaptability our number one priority through the use of _Exchanges_.
+Lets run through how this works by creating our own Dedup exchange - a middleware to debounce incoming query operations.
+
+> Note: We've already provided a bunch of exchanges that we thought you would find
```

Comment:
This will need to point to the basics section, not the source code. I assume people can find the source rather easy but will actually want to know which ones are standard first

Reply:
Trying to do this on my phone on the tube - I swear GitHub had a quick fix option added 😕